### ISSUE TITLE
AOT Support

### ISSUE THREAD

**Issue by ghost1372** [microsoft/WinUI-Gallery#1627]
### Which version of the app?

WinUI 3 Gallery

### Description

I am looking to see how AOT is implemented in the gallery. I didn't see anything related to AOT, has it not started yet?
I'm more interested to see how [this line](https://github.com/microsoft/WinUI-Gallery/blob/b8bbce3c9073b9ada0302a88640a5c04e361eb47/WinUIGallery/DataModel/ControlInfoDataSource.cs#L202) will be modified.

https://github.com/microsoft/WinUI-Gallery/blob/b8bbce3c9073b9ada0302a88640a5c04e361eb47/WinUIGallery/DataModel/ControlInfoDataSource.cs#L202

**Comment by karkarl** [comment]
We are working on the AOT samples for WinUI-Gallery. Unfortunately, there are unforeseen issues that is causing the PR validation pipeline to be blocked. Hopefully this can be resolved soon and we can have the samples in!

**Comment by marcelwgn** [comment]
Thank you for your elaborate comment and sorry for the delayed response on my part. I think your approach is good, especially since it makes it easy to keep maintaining that list. If you would like to create a PR for this, I am sure we would accept such a contribution!

**Comment by ghost1372** [comment]
> Thank you for your elaborate comment and sorry for the delayed response on my part. I think your approach is good, especially since it makes it easy to keep maintaining that list. If you would like to create a PR for this, I am sure we would accept such a contribution!

Thank you! @marcelwgn  
I'm planning to start working on a PR.
Should I wait until this one is merged https://github.com/microsoft/WinUI-Gallery/pull/1908 or can I go ahead and begin?

**Comment by niels9001** [comment]
> > Thank you for your elaborate comment and sorry for the delayed response on my part. I think your approach is good, especially since it makes it easy to keep maintaining that list. If you would like to create a PR for this, I am sure we would accept such a contribution!
> 
> Thank you! @marcelwgn  
> I'm planning to start working on a PR.
> Should I wait until this one is merged https://github.com/microsoft/WinUI-Gallery/pull/1908 or can I go ahead and begin?
> 

Maybe branch of #1908 and I'll update that branch with removing the WinUI TitleBar due to blocking bugs, and add a custom TitleBar for now.

**Comment by marcelwgn** [comment]
The only overlap between this and #1908 is the fact that some control in that PR calls navigate right? If we are removing the WinUI Titlebar from that PR (which essentially is maintaing status quo anyway) then I think @niels9001 approach is good. But i also hope we can get that in soon because code structure wise it is a big improvement and the only overlap between this issue an #1908 is on how to call navigate right?

**Comment by ghost1372** [comment]
Hi @marcelwgn @niels9001,
I’ve added initial AOT support
Thanks!

**Comment by ghost1372** [comment]
Native AOT PR Merged and this issue can be closed

**Comment by marcelwgn** [comment]
I think we need to reopen this since the navigation is still using "magic" type strings and is failing in NAOT.

**Comment by ghost1372** [comment]
Hi @marcelwgn 
So we’ll have to take on the hard work. Is there a deadline for this? I’ll be busy for the next few days.

**Comment by marcelwgn** [comment]
This is not urgent at all so no worries. And please dont feel obliged to work on this, this is also on my list of things I'm just not sure when I can start working on it. If I start working on it, I will let you know so there is no duplicated work.

**Comment by ghost1372** [comment]
Hi @marcelwgn 
Today I wrote a source generator that can easily extract uniqueid from JSON and generate a mapping class. However, it looks like we need to make some changes to the JSON file and the navigation method. Currently, the JSON file only has a simplified uniqueid, and in the generated code the type isn’t recognized. 

<img width="1107" height="663" alt="Image" src="https://github.com/user-attachments/assets/7529a836-fe91-4d2d-ae40-fc0bbfeedaa1" />

So, we need to standardize the uniqueid to include the exact page namespace plus the filename.
For example: `WinUIGallery.ControlPages.XamlResourcesPage`, then generator can work as expected and navigate method can navigate easily.
 This approach is better than the T4 template, which is no longer supported.

**Comment by ghost1372** [comment]
Hi @marcelwgn 
Can we close it now?

**Comment by marcelwgn** [comment]
Yes, I think we can close it for now. Thank you for all your work on this @ghost1372 !
